File: Algorithms/744.Find-Smallest-Letter-Greater-Than-Target/find_smallest_letter_greater_than_target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        length = len(letters)
        start = 0
        end = length - 1
        while start <= end:
            mid = (start + end) // 2
            if letters[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
        return  letters[0] if start >= length else letters[start]

File: Algorithms/744.Find-Smallest-Letter-Greater-Than-Target/find_smallest_letter_greater_than_target.go
func nextGreatestLetter(letters []byte, target byte) byte {
    length := len(letters)
    start := 0
    end := length - 1
    for start <= end {
        mid := (start + end) / 2
        if target < letters[mid] {
            end = mid -1
        } else {
            start = mid + 1
        }
    }
    if start >= length {
        return letters[0]
    } else {
        return letters[start]
    }
}
